alphastats/DataSet_Plot.py

```diff
@@ -125,7 +129,20 @@ def _plot_dimensionality_reduction(self, group, method, circle, **kwargs):
             # TODO implement UMAP??
             return
 
-        fig = px.scatter(components, x=0, y=1, labels=labels, color=group_color,)
+        components = pd.DataFrame(components)
+        components["sample"] = sample_names
+
+        fig = px.scatter(components, x=0, y=1, labels=labels, color=group_color, hover_data=[components["sample"]])
+
+        # rename hover_data_0 to sample
+        fig_dict = fig.to_plotly_json()
+        data = fig_dict.get("data")
+        
+        for count, d in enumerate(data):
+            hover = d.get("hovertemplate").replace("hover_data_0", "sample")
```

Comment:
This looks a bit like a hack.

You should be able to update the hover-text like so:

```
fig.update_traces(
    hovertemplate="<br>".join([
        "ColX: %{x}",
        "ColY: %{y}",
        "Col1: %{customdata[0]}",
        "Col2: %{customdata[1]}",
        "Col3: %{customdata[2]}",
    ])
)
```
See also: 
https://stackoverflow.com/questions/59057881/python-plotly-how-to-customize-hover-template-on-with-what-information-to-show